packages/vue-components/src/panels/PanelBase.js

```diff
@@ -99,76 +99,85 @@ export default {
       localExpanded: false,
       localMinimized: false,
       wasRetrieverLoaded: false,
+      expandedPanelHeight: 0, // to be set dynamically
+      collapsedPanelHeight: 0, // will be >0 if show-preview mode
     };
   },
   methods: {
     toggle() {
-      if (!this.wasRetrieverLoaded && !this.localExpanded) {
-        /*
-        Let the retriever load before toggling localExpanded which triggers the animation,
-        as we need to know the scrollHeight of the content for the animation to work.
-        */
-        this.wasRetrieverLoaded = true;
-        this.$nextTick(() => {
-          this.localExpanded = true;
-        });
+      if (this.localExpanded) {
+        // COLLAPSE
+        // panel maxHeight is 'none' at this moment (as event listener set it to 'none' after expansion)
+        // thus, we need to reset the maxHeight to its current height (for collapse transition to work)
+        this.$refs.panel.style.maxHeight = `${this.$refs.panel.scrollHeight}px`;
+        // we need to reset / 'memorize' the expandedPanelHeight everytime we collapse
+        // this is because we may have expanded nested panels and the parent panel height is now different
+        this.expandedPanelHeight = this.$refs.panel.scrollHeight;
+        // need to wait for DOM update, after resetting panel maxHeight
+        setTimeout(() => {
+          // transition to collapsed panel
+          this.$refs.panel.style.maxHeight = `${this.collapsedPanelHeight}px`;
+        }, 0); // nextTick doesn't work, not sure why :O
       } else {
-        this.localExpanded = !this.localExpanded;
+        // EXPAND
+        this.$refs.panel.style.maxHeight = `${this.expandedPanelHeight}px`;
```

Comment:
you should be able to reference `this.$refs.panel.scrollHeight` here directly as well